akka-projection-r2dbc/src/main/scala/akka/projection/r2dbc/internal/R2dbcOffsetStore.scala

```diff
@@ -630,15 +652,19 @@ private[projection] class R2dbcOffsetStore(
 
         val offsetInserts = dao.insertTimestampOffsetInTx(conn, filteredRecords)
 
-        offsetInserts.map { _ =>
+        offsetInserts.flatMap { _ =>
```

Comment:
This was the bug. I think it was introduced in a recent version when the lazy loading of offsets was added.

Reply:
👍🏼  Not noticed in #1255